### DISCUSSION TITLE
Realistic IMU Sensor simulation

### DISCUSSION THREAD

**Discussion by RaviBeagle** [carla-simulator/carla#5854] [Q&A]
Hello carla-community,

How can set noise and bias values so that I can have a more real world IMU type of sensor simulation.

```
Blueprint attribute	Type	Default	Description
noise_accel_stddev_x	float	0	Standard deviation parameter in the noise model for acceleration (X axis).
noise_accel_stddev_y	float	0	Standard deviation parameter in the noise model for acceleration (Y axis).
noise_accel_stddev_z	float	0	Standard deviation parameter in the noise model for acceleration (Z axis).
noise_gyro_bias_x	float	0	Mean parameter in the noise model for the gyroscope (X axis).
noise_gyro_bias_y	float	0	Mean parameter in the noise model for the gyroscope (Y axis).
noise_gyro_bias_z	float	0	Mean parameter in the noise model for the gyroscope (Z axis).
noise_gyro_stddev_x	float	0	Standard deviation parameter in the noise model for the gyroscope (X axis).
noise_gyro_stddev_y	float	0	Standard deviation parameter in the noise model for the gyroscope (Y axis).
noise_gyro_stddev_z	float	0	Standard deviation parameter in the noise model for the gyroscope (Z axis).
noise_seed	int	0	Initializer for a pseudorandom number generator.
```
With the default values we will get a most accurate IMU without drift!

But for realistic simulation we would need a proper noise model ?

**Comment by goodgoodliving** [comment]
> Hello carla-community,
> 
> How can set noise and bias values so that I can have a more real world IMU type of sensor simulation.
> 
> ```
> Blueprint attribute	Type	Default	Description
> noise_accel_stddev_x	float	0	Standard deviation parameter in the noise model for acceleration (X axis).
> noise_accel_stddev_y	float	0	Standard deviation parameter in the noise model for acceleration (Y axis).
> noise_accel_stddev_z	float	0	Standard deviation parameter in the noise model for acceleration (Z axis).
> noise_gyro_bias_x	float	0	Mean parameter in the noise model for the gyroscope (X axis).
> noise_gyro_bias_y	float	0	Mean parameter in the noise model for the gyroscope (Y axis).
> noise_gyro_bias_z	float	0	Mean parameter in the noise model for the gyroscope (Z axis).
> noise_gyro_stddev_x	float	0	Standard deviation parameter in the noise model for the gyroscope (X axis).
> noise_gyro_stddev_y	float	0	Standard deviation parameter in the noise model for the gyroscope (Y axis).
> noise_gyro_stddev_z	float	0	Standard deviation parameter in the noise model for the gyroscope (Z axis).
> noise_seed	int	0	Initializer for a pseudorandom number generator.
> ```
> 
> With the default values we will get a most accurate IMU without drift!
> 
> But for realistic simulation we would need a proper noise model ?

@RaviBeagle Have you resolved it? I'm encountering the same issue now.

**Reply by aminemoussi** [reply]
me too
